amaznot/app/Http/Controllers/OrderController.php

```diff
@@ -112,4 +115,29 @@ private function getOrderItems(Request $request, $orderId)
 
         return $orderItems;
     }
+    
+    public function orderDetails(Request $request, $id)
+    {
+        //Check credentials
+        if ($redirect = parent::redirectOnNotUser($request))
+        {
+            return $redirect;
+        }
+
+        $orderDetails = OrderItem::with('order_items')->where('id', $id)->get();
+        $ordersDetails = OrderItem::paginate(10);
```

Comment:
This has the same issue as above where the second query overwrites the first. Additionally, the first query is incorrect as well. It only returns the first order item where the id matches (you're matching the $id variable to the 'id' column instead of to the 'order_id' column). Also, the purpose of querying to the orderitems table is to obtain a list of product ids, so that you can display the products. No query to the product table is made here.

Reply:
Thought it was a mistake when it presented everything with the same orders id's. But i'll fix it so it shows everything with the same order ID